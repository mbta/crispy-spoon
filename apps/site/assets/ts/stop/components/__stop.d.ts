import {
  Direction,
  DirectionId,
  EnhancedRoute,
  Stop,
  Alert
} from "../../__v3api";
import { MapData } from "../../leaflet/components/__mapdata";

export interface TypedRoutes {
  group_name: string;
  routes: RouteWithDirections[];
}

export interface StopMapData {
  map_data: MapData;
  map_srcset: string;
  map_url: string;
}

export interface RouteWithDirection {
  direction_id: DirectionId | null;
  route: EnhancedRoute;
}

export interface SuggestedTransfer {
  stop: Stop;
  distance: number;
  routes_with_direction: RouteWithDirection[];
}

export interface StopPageData {
  stop: Stop;
  street_view_url: string | null;
  routes: TypedRoutes[];
  routes_with_direction?: RouteWithDirection[];
  tabs: Tab[];
  zone_number: string;
  alerts: Alert[];
  suggested_transfers: SuggestedTransfer[];
  retail_locations: RetailLocationWithDistance[];
  tab: string;
  alerts_tab: AlertsTab;
  routes_and_alerts: { [key: string]: Alert[] };
}

interface AlertData {
  alerts: Alert[];
  empty_message: string;
}

export interface AlertsTab {
  [key: string]: AlertData | string;
  upcoming: AlertData;
  current: AlertData;
  all: AlertData;
  initial_selected: string;
}

export interface Tab {
  badge?: TabBadge;
  class: string;
  href: string;
  id: string;
  name: string;
  selected: string;
}

export interface TabBadge {
  content: string;
  aria_label: string;
  class?: string;
}

export interface RetailLocation {
  address: string;
  latitude: number;
  longitude: number;
  name: string;
  payment: string[];
  phone: string;
}

export interface RetailLocationWithDistance {
  distance: string;
  location: RetailLocation;
}

export interface RouteWithDirections {
  route: EnhancedRoute;
  directions: Direction[];
}
