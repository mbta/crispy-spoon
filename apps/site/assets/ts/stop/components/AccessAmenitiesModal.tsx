import React from "react";
import { Alert, Facility } from "../../__v3api";
import Alerts from "../../components/Alerts";
import { hasFacilityAlert } from "../../models/alert";
import { AmenityModal, AmenityLink } from "./amenities/AmenityCard";

const AccessAmenitiesModal = ({
  stopName,
  alerts,
  facilities,
  facilityType
}: {
  stopName: string;
  alerts: Alert[];
  facilities: Facility[];
  facilityType: "Elevators" | "Escalators";
}): JSX.Element => {
  const hasFacilities = facilities ? facilities.length > 0 : false;
  const linkText =
    facilityType === "Elevators"
      ? "Report an elevator issue"
      : "Report an escalator issue";
  return (
    <>
      {hasFacilities && (
        <AmenityModal headerText={`${facilityType} at ${stopName}`}>
          <Alerts alerts={alerts} />
          <h2>Elevator Status</h2>

          <table className="access-amenities-table">
            <tbody>
              <tr className="access-amenities-header access-amenities-row">
                <th className="access-amenities-name">Elevator</th>
                <th className="status">Status</th>
              </tr>
              {facilities?.map(facility => {
                return (
                  <tr
                    key={facility.id}
                    className="access-amenities-row access-amenities-text"
                  >
                    <td className="access-amenities-name">
                      {facility.attributes.short_name}
                    </td>
                    {hasFacilityAlert(facility.id, alerts) ? (
                      <td className="status">
                        <i className="fa-solid fa-circle amenity-status amenity-out" />
                        Out of Order
                      </td>
                    ) : (
                      <td className="status">
                        <i className="fa-solid fa-circle amenity-status amenity-working" />
                        Working
                      </td>
                    )}
                  </tr>
                );
              })}
            </tbody>
          </table>
          <AmenityLink
            url="/accessibility/trip-planning"
            text="Plan an accessible trip"
          />
          <AmenityLink url="/customer-support" text={linkText} />
        </AmenityModal>
      )}
    </>
  );
};
export default AccessAmenitiesModal;
