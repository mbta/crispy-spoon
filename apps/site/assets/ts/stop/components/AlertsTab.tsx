import React, { useState, ReactElement } from "react";
import { AlertsTab as AlertsTabType, AlertData } from "./__stop";
import Alerts from "../../components/Alerts";
import { handleReactEnterKeyPress } from "../../helpers/keyboard-events";

interface Props {
  alertsTab: AlertsTabType;
}

const selectedClass = (filter: string, selected: string): string =>
  selected === filter ? "m-alerts__time-filter--selected" : "";

const timeFilter = (
  name: string,
  title: string,
  currentFilter: string,
  onClick: React.MouseEventHandler
): ReactElement<HTMLElement> => (
  <div
    role="button"
    id={name}
    tabIndex={0}
    className={`m-alerts__time-filter ${selectedClass(name, currentFilter)}`}
    onClick={onClick}
    onKeyPress={e => handleReactEnterKeyPress(e, onClick)}
  >
    {title}
  </div>
);

const AlertsTab = ({ alertsTab }: Props): ReactElement<HTMLElement> => {
  // eslint-disable-next-line camelcase
  const initialFilter: string = alertsTab.initial_selected || "all";
  const [filter, changeFilter] = useState(initialFilter);

  return (
    <div className="container">
      <div className="page-section station">
        <div className="page-section">
          <div className="row">
            <div className="col-xs-12 col-lg-3">
              <h3 className="h3">Filter by type</h3>
              <div className="m-alerts__time-filters">
                {timeFilter("all", "All Alerts", filter, () =>
                  changeFilter("all")
                )}
                {timeFilter("current", "Current Alerts", filter, () =>
                  changeFilter("current")
                )}
                {timeFilter("upcoming", "Planned Service Alerts", filter, () =>
                  changeFilter("upcoming")
                )}
              </div>
            </div>
            <div className="col-xs-12 col-lg-8 col-lg-offset-1">
              <h2 className="h2">Alerts</h2>
              <div>
                {/* eslint-disable */}
                {(alertsTab[filter] as AlertData).alerts.length ? (
                  <Alerts alerts={(alertsTab[filter] as AlertData).alerts} />
                ) : (
                  <div className="callout">
                    {(alertsTab[filter] as AlertData).empty_message}
                  </div>
                )}
                {/* eslint-enable */}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default AlertsTab;
