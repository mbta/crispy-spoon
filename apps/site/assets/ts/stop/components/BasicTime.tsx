import React from "react";
import parse from "date-fns/parse";
import { formatRelativeTime, formatToBostonTime } from "../../helpers/date";

export interface BasicTimeProps {
  displayType: "relative" | "absolute";
  strikethrough?: boolean;
  time: Date | undefined;
  targetDate?: Date;
}
const BasicTime = ({
  displayType,
  strikethrough,
  time,
  targetDate
}: BasicTimeProps): JSX.Element | null => {
  if (!time) return null;

  const formattedTime =
    displayType === "relative"
      ? formatRelativeTime(time, targetDate)
      : formatToBostonTime(time, "h:mm aa");

  // displayType === "relative" might produce an absolute-formatted time if it's
  // more than an hour away. parse the result to find the truth.
  const parsedTime = parse(formattedTime, "h:mm aa", new Date()).getTime();
  const isAbsolute = Number.isNaN(parsedTime) ? "relative" : "absolute";
  // add data attribute to be used for later manipulation and filtering
  const dataType: "relative" | "absolute" | "imminent" =
    formattedTime === "<1 minute away" ? "imminent" : isAbsolute;
  return (
    <time
      data-type={dataType}
      dateTime={time.toISOString()}
      className={strikethrough ? "strikethrough" : ""}
    >
      {formattedTime}
    </time>
  );
};

export default BasicTime;
