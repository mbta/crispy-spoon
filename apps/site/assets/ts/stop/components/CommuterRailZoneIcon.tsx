import React, { ReactElement } from "react";
import { clickRoutePillAction, Dispatch } from "../state";

const crZone = (
  zoneNumber?: string,
  dispatch?: Dispatch
): ReactElement<HTMLElement> | false => {
  const content = `Zone ${zoneNumber}`;

  let tag = (
    <div className="m-stop-page__header-feature">
      <span className="m-stop-page__icon c-icon__cr-zone">{content}</span>
    </div>
  );

  if (dispatch) {
    tag = (
      <a
        href="#route-card-list"
        onClick={() =>
          dispatch && dispatch(clickRoutePillAction("commuter_rail"))
        }
        className="m-stop-page__header-feature"
      >
        <span className="m-stop-page__icon c-icon__cr-zone">{content}</span>
      </a>
    );
  }

  return !!zoneNumber && zoneNumber.length > 0 && tag;
};

const CommuterRailZoneIcon = ({
  zoneNumber,
  dispatch
}: {
  zoneNumber?: string;
  dispatch?: Dispatch;
}): ReactElement<HTMLElement> => {
  return <>{crZone(zoneNumber, dispatch)}</>;
};

export { CommuterRailZoneIcon as default };
