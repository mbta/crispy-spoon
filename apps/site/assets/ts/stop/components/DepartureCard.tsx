import { reject } from "lodash";
import React, { ReactElement } from "react";
import { Alert, Route } from "../../__v3api";
import { routeName, routeToModeIcon } from "../../helpers/route-headers";
import { routeBgClass } from "../../helpers/css";
import renderSvg from "../../helpers/render-svg";
import useDepartureRow from "../../hooks/useDepartureRow";
import { DepartureInfo } from "../../models/departureInfo";
import { allAlertsForDirection } from "../../models/alert";
import { departureInfoInRoutePatterns } from "../../helpers/departureInfo";
import { isACommuterRailRoute, isSubwayRoute } from "../../models/route";
import DepartureTimes from "./DepartureTimes";
import {
  RoutePatternGroup,
  RoutePatternWithPolyline,
  sortedGroupedRoutePatterns
} from "../../models/route-patterns";

const isNoncanonicalAndNoDepartures = (
  routePatterns: RoutePatternWithPolyline[],
  departures: DepartureInfo[]
): boolean => {
  const isNonCanonical = !routePatterns.find(rp => !!rp.canonical);
  return isNonCanonical && departures.length === 0;
};

const DepartureCard = ({
  alertsForRoute,
  departuresForRoute,
  routePatternsByHeadsign,
  route
}: {
  alertsForRoute: Alert[];
  departuresForRoute: DepartureInfo[];
  routePatternsByHeadsign: RoutePatternGroup;
  route: Route;
}): ReactElement<HTMLElement> | null => {
  const { setRow } = useDepartureRow([route]);

  let sortedRoutePatternsByHeadsign = sortedGroupedRoutePatterns(
    routePatternsByHeadsign
  );

  if (isSubwayRoute(route)) {
    // remove certain headsigns based on route pattern and departures
    sortedRoutePatternsByHeadsign = reject(
      sortedRoutePatternsByHeadsign,
      entry => {
        const [, { route_patterns: routePatterns }] = entry;
        const departures = departuresForRoute.filter(d =>
          departureInfoInRoutePatterns(d, routePatterns)
        );
        return isNoncanonicalAndNoDepartures(routePatterns, departures);
      }
    );
    // don't render a route card if there's no headsigns left to show
    if (sortedRoutePatternsByHeadsign.length === 0) return null;
  }

  return (
    <li className="departure-card">
      <a
        className={`departure-card__route ${routeBgClass(route)} notranslate`}
        href={`/schedules/${route.id}`}
        data-turbolinks="false"
      >
        {renderSvg("c-svg__icon", routeToModeIcon(route), true)}{" "}
        {routeName(route)}
      </a>
      {sortedRoutePatternsByHeadsign.map(
        ([
          headsign,
          { direction_id: directionId, route_patterns: routePatterns }
        ]) => {
          const onClick = (): void =>
            setRow({
              routeId: route.id,
              directionId: directionId.toString(),
              headsign
            });

          return (
            <DepartureTimes
              key={headsign}
              alertsForDirection={allAlertsForDirection(
                alertsForRoute,
                directionId
              )}
              headsign={headsign}
              departures={departuresForRoute.filter(d =>
                departureInfoInRoutePatterns(d, routePatterns)
              )}
              onClick={onClick}
              isCR={isACommuterRailRoute(route)}
              hasService={routePatterns.length !== 0}
            />
          );
        }
      )}
    </li>
  );
};

export default DepartureCard;
