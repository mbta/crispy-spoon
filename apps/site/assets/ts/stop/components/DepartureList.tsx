import React, { ReactElement } from "react";
import { filter } from "lodash";
import { Alert, DirectionId, Route, Stop, Trip } from "../../__v3api";
import { DepartureInfo } from "../../models/departureInfo";
import { SUBWAY, departuresListFromInfos } from "../../helpers/departureInfo";
import { routeBgClass } from "../../helpers/css";
import { routeName, routeToModeIcon } from "../../helpers/route-headers";
import renderSvg from "../../helpers/render-svg";
import { isSuppressiveAlert } from "../../models/alert";
import Alerts from "../../components/Alerts";
import { isACommuterRailRoute } from "../../models/route";

interface DepartureListProps {
  route: Route;
  stop: Stop;
  departures: DepartureInfo[];
  directionId: DirectionId;
  headsign: string;
  alerts: Alert[];
  hasService: boolean;
  targetDate?: Date | undefined;
}

const displayNoUpcomingTrips = (
  message = "No more trips today"
): JSX.Element => {
  return (
    <div className="c-alert-item--low m-8 d-flex justify-content-center align-items-center pb-40 pt-40">
      {message}
    </div>
  );
};

const DepartureList = ({
  route,
  stop,
  departures,
  directionId,
  headsign,
  alerts,
  hasService,
  targetDate
}: DepartureListProps): ReactElement<HTMLElement> => {
  const isCR = isACommuterRailRoute(route);

  // don's show cancelled departures for subway
  const modeSpecificDepartures: DepartureInfo[] = filter(
    departures,
    (d: DepartureInfo) => !(d.isCancelled && d.routeMode === SUBWAY)
  );
  const numPredictions = modeSpecificDepartures.filter(d => !!d.prediction)
    .length;
  const alertsShouldSuppressDepartures = alerts.some(alert =>
    isSuppressiveAlert(alert, numPredictions)
  );
  const tripForSelectedRoutePattern: Trip | undefined =
    modeSpecificDepartures[0]?.trip;

  const noTrips =
    modeSpecificDepartures.length === 0 && displayNoUpcomingTrips();
  const noService = !hasService && displayNoUpcomingTrips("No trips today");
  const noServiceOrNoTrips = noService || noTrips;

  return (
    <>
      <div className="stop-departures departure-list-header">
        <div className={`departure-card__route ${routeBgClass(route)}`}>
          <div className="notranslate">
            {renderSvg("c-svg__icon", routeToModeIcon(route), true)}{" "}
            {routeName(route)}
          </div>
          <a
            className="open-schedule"
            href={`/schedules/${route.id}/line?schedule_direction[direction_id]=${directionId}&schedule_direction[variant]=${tripForSelectedRoutePattern?.route_pattern_id}&schedule_finder[direction_id]=${directionId}&schedule_finder[origin]=${stop.id}`}
            data-turbolinks="false"
          >
            View all schedules
          </a>
        </div>
      </div>
      <h2 className="departure-list__sub-header">
        <div className="departure-list__origin-stop-name notranslate">
          {stop.name} to
        </div>
        <div className="departure-list__headsign notranslate">{headsign}</div>
      </h2>
      {alerts.length ? <Alerts alerts={alerts} /> : null}
      {noServiceOrNoTrips ||
        (!alertsShouldSuppressDepartures && (
          <ul className="stop-routes__departures list-unstyled">
            {departuresListFromInfos(modeSpecificDepartures, isCR, targetDate)}
          </ul>
        ))}
    </>
  );
};

export default DepartureList;
