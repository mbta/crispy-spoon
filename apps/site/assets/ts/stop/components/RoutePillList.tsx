import React, { ReactElement } from "react";
import { EnhancedRoute, Route } from "../../__v3api";
import { TypedRoutes } from "./__stop";
import { breakTextAtSlash } from "../../helpers/text";
import { routeToCSSClass } from "../../helpers/css";
import PillLink from "./PillLink";

interface RoutePillListProps {
  routes: TypedRoutes[];
  showGroupName?: boolean;
}

interface RoutePillProps {
  route: Route | EnhancedRoute;
}

const busName = (name: string): string =>
  name.startsWith("SL") ? "silver-line" : "bus";

const modeNameForBg = ({ name, type }: EnhancedRoute | Route): string => {
  switch (type) {
    case 0:
    case 1:
      return routeToCSSClass(name);
    case 2:
      return "commuter-rail";
    case 4:
      return "ferry";
    default:
      return busName(name);
  }
};

export const modeBgClass = (route: EnhancedRoute | Route): string =>
  `u-bg--${modeNameForBg(route)}`;

const RoutePill = ({ route }: RoutePillProps): ReactElement<HTMLElement> => (
  <PillLink
    displayText={breakTextAtSlash(route.name)}
    linkURL={`/schedules/${route.id}`}
    optionalCSS={modeBgClass(route)}
  />
);

const RoutePillList = ({
  routes,
  showGroupName
}: RoutePillListProps): ReactElement<HTMLElement> => (
  <div className="m-route-pills">
    {routes.map(typedRoute => (
      <div
        key={typedRoute.group_name}
        className={`m-route-pills--${typedRoute.group_name}`}
      >
        {showGroupName && <h4 className="h4">{typedRoute.group_name}</h4>}
        {typedRoute.routes.map(routeWithDirections => (
          <RoutePill
            key={routeWithDirections.route.id}
            route={routeWithDirections.route}
          />
        ))}
      </div>
    ))}
  </div>
);

export default RoutePillList;
