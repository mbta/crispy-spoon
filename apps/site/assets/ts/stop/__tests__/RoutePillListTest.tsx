import React from "react";
import renderer from "react-test-renderer";
import stopData from "./stopData.json";
import { StopPageData, TypedRoutes } from "../components/__stop";
import RoutePillList from "../components/RoutePillList";

const data = JSON.parse(JSON.stringify(stopData)) as StopPageData;

it("renders", () => {
  document.body.innerHTML =
    '<div><div id="react-root"><div id="test"></div></div></div>';

  const tree = renderer.create(<RoutePillList routes={data.routes} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders a ferry route", () => {
  /* eslint-disable camelcase */
  const routes: TypedRoutes[] = [
    {
      group_name: "ferry",
      routes: [
        {
          route: {
            type: 4,
            name: "Charlestown Ferry",
            header: "Charlestown Ferry",
            long_name: "Charlestown Ferry",
            id: "Boat-F4",
            direction_names: { "0": "Outbound", "1": "Inbound" },
            direction_destinations: { "0": "Charlestown", "1": "Long Wharf" },
            description: "ferry",
            alerts: []
          },
          directions: []
        }
      ]
    }
  ];
  /* eslint-enable camelcase */

  const tree = renderer.create(<RoutePillList routes={routes} />).toJSON();

  expect(tree).toMatchSnapshot();
});
