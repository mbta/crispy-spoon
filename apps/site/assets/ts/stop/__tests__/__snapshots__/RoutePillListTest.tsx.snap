// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="m-route-pills"
>
  <div
    className="m-route-pills--subway"
  >
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--red-line
    "
      href="/schedules/Red"
      rel="noreferrer"
      target="_self"
    >
      Red Line
    </a>
  </div>
  <div
    className="m-route-pills--bus"
  >
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--silver-line
    "
      href="/schedules/741"
      rel="noreferrer"
      target="_self"
    >
      SL1
    </a>
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--silver-line
    "
      href="/schedules/742"
      rel="noreferrer"
      target="_self"
    >
      SL2
    </a>
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--silver-line
    "
      href="/schedules/743"
      rel="noreferrer"
      target="_self"
    >
      SL3
    </a>
  </div>
  <div
    className="m-route-pills--commuter_rail"
  >
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
      href="/schedules/CR-Fairmount"
      rel="noreferrer"
      target="_self"
    >
      Fairmount Line
    </a>
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
      href="/schedules/CR-Worcester"
      rel="noreferrer"
      target="_self"
    >
      Framingham/​Worcester Line
    </a>
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
      href="/schedules/CR-Franklin"
      rel="noreferrer"
      target="_self"
    >
      Franklin Line
    </a>
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
      href="/schedules/CR-Greenbush"
      rel="noreferrer"
      target="_self"
    >
      Greenbush Line
    </a>
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
      href="/schedules/CR-Kingston"
      rel="noreferrer"
      target="_self"
    >
      Kingston/​Plymouth Line
    </a>
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
      href="/schedules/CR-Middleborough"
      rel="noreferrer"
      target="_self"
    >
      Middleborough/​Lakeville Line
    </a>
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
      href="/schedules/CR-Needham"
      rel="noreferrer"
      target="_self"
    >
      Needham Line
    </a>
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
      href="/schedules/CR-Providence"
      rel="noreferrer"
      target="_self"
    >
      Providence/​Stoughton Line
    </a>
  </div>
</div>
`;

exports[`renders a ferry route 1`] = `
<div
  className="m-route-pills"
>
  <div
    className="m-route-pills--ferry"
  >
    <a
      className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--ferry
    "
      href="/schedules/Boat-F4"
      rel="noreferrer"
      target="_self"
    >
      Charlestown Ferry
    </a>
  </div>
</div>
`;
